logic/scanning_logic.py

```diff
@@ -0,0 +1,629 @@
+# -*- coding: utf-8 -*-
+"""
+This module operates a confocal microsope.
+
+Qudi is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Qudi is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Qudi. If not, see <http://www.gnu.org/licenses/>.
+
+Copyright (c) the Qudi Developers. See the COPYRIGHT.txt file at the
+top-level directory of this distribution and at <https://github.com/Ulm-IQO/qudi/>
+"""
+
+from qtpy import QtCore
+from collections import OrderedDict
+from itertools import combinations
+import time
+import datetime
+import numpy as np
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import copy
+
+from logic.generic_logic import GenericLogic
+from core.util.mutex import Mutex
+from core.module import Connector, ConfigOption, StatusVar
+
+
+class ScanData:
+    """
+
+    """
+    def __init__(self, scan_axes, channel_config, scanner_settings):
+        self.timestamp = datetime.datetime.now()
+        self._scan_axes = tuple(scan_axes)
+        if self._scan_axes not in scanner_settings['scan_axes']:
+            raise ValueError('scan_axes must be tuple of axes name strings contained in '
+                             'scanner_settings')
+        self._target_ranges = tuple(scanner_settings['scan_range'][ax] for ax in self._scan_axes)
+        self._resolution = tuple(scanner_settings['scan_resolution'][ax] for ax in self._scan_axes)
+        self._channel_names = tuple(channel_config)
```

Comment:
I don't think default values are applicable here. These parameters are mandatory for a scan and so it's fine to throw an error if they are absent.

Regarding the use of tuples:
Since all these parameters should represent a snapshot during __init__ and must stay the same as long as this instance exists, I found it suitable to copy the values into a non-mutable data type. This would also avoid possible errors in the future if the dict values are mutable data types.
In addition, having a couple of instance attributes increases readability compared to a single dict in my opinion.